kong/db/schema/entities/certificates.lua

```diff
@@ -29,6 +36,21 @@ return {
 
         return true
       end,
-    } }
+    } },
+    { custom_entity_check = {
+      field_sources = { "cert_alt", "key_alt" },
+      fn = function(entity)
+        if type(entity.cert_alt) == "string" and type(entity.key_alt) == "string" then
+          local cert_alt = openssl_x509.new(entity.cert_alt)
+          local key_alt = openssl_pkey.new(entity.key_alt)
+
```

Comment:
Other types of check we should do because this feature could easily be used incorrectly for non-suspecting users:

1. The hash of `cert_alt` and `cert` differs. This can be obtained using https://github.com/fffonion/lua-resty-openssl#x509digest. This ensures user does not accidentally put both fields to the same cert because they may thought both are required.
2. The type of public key for `cert_alt` and `cert` differs. I think this could be done using `x509:get_pkey():get_key_type` and check the `.id` field of the returned object differs. (@fffonion does this looks correct to you?)
This ensures user does not accidentally put both fields to the same cert type (for example, put two RSA certs of different expiration date and hoping it would work).

Reply:
@dndx, I have now added the `distinct` validation and test, but I needed to add this too: https://github.com/Kong/kong/pull/6536/commits/93d53663c91412929c4a5e5faa433da0bd6d460d

Not sure if it is such a big deal, if it is, then I need to change the tests. I think running `distinct` on invalid fields is not needed, what do you think? The value of it is that we can report the error at the same time as the other errors, I think.

Reply:
Distinct validator was now replaced with custom validation.